mautic/mautic
# Add lock support on maintenances commands

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR added lock for 
- `mautic:maintenance:cleanup`
- `mautic:unusedip:delete`

It prevent run same command twice or more times.



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.  Run  command and try it run twice in same time

- `mautic:maintenance:cleanup`
- `mautic:unusedip:delete`

3. You should see `Script in progress. Can force execution by using --bypass-locking` on second call.
